api/src/main/java/ca/bc/gov/educ/api/school/service/SchoolService.java

```diff
@@ -0,0 +1,55 @@
+package ca.bc.gov.educ.api.school.service;
+
+import ca.bc.gov.educ.api.school.exception.EntityNotFoundException;
+import ca.bc.gov.educ.api.school.model.*;
+import ca.bc.gov.educ.api.school.repository.SchoolRepository;
+import lombok.Getter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+import static lombok.AccessLevel.PRIVATE;
+
+@Service
+public class SchoolService {
+  private static final String MINCODE_ATTRIBUTE = "minCode";
+  private static final int DISTNO_LENGTH = 3;
+
+  /**
+   * The School repository.
+   */
+  @Getter(PRIVATE)
+  private final SchoolRepository schoolRepository;
+
+  /**
+   * Instantiates a new School service.
+   *
+   * @param schoolRepository  School repository
+   */
+  @Autowired
+  public SchoolService(SchoolRepository schoolRepository) {
+    this.schoolRepository = schoolRepository;
+  }
+
+  /**
+   * Search for SchoolEntity by MinCode
+   *
+   * @param minCode the unique minCode for a given school.
+   * @return the School entity if found.
+   */
+  public SchoolEntity retrieveSchoolByMinCode(String minCode) {
+    Optional<SchoolEntity> result = Optional.empty();
+    if(minCode.length() > DISTNO_LENGTH) {
```

Comment:
Does minCode need a null check?

Reply:
minCode is required by the controller, so it should not be null.